File: 백준/Gold/5430. AC/AC.py
import sys
from collections import deque
# sys.stdin = open("input.txt",'r')
# sys.stdout = open("output.txt",'w')
input = sys.stdin.readline
MIS = lambda: map(int, input().rstrip().split())

def print_arr(arr):
    print("[", end="")
    print(*arr, sep=",", end="")
    print("]")

case_cnt = int(input())
arr_len = 0

for _ in range(case_cnt):
    is_abrupt = False
    is_reverse = False

    funcs = input()
    arr_len = int(input())
    arr = input()
    arr = arr.replace('[',' ')
    arr = arr.replace(']',' ')
    arr =arr.replace(',',' ')

    arr = deque(arr.split())
    
    for func in list(funcs):
        
        if not arr:
            if func == "D":
                is_abrupt = True
                break

        if func == "R":
            is_reverse = not is_reverse
        elif func == "D":
            if is_reverse:
                arr.pop()
            else:
                arr.popleft()
    
    if is_abrupt:
        print("error")
        continue
    if is_reverse: arr.reverse()
    print_arr(arr)

    

File: 백준/Gold/5430. AC/AC.cc
#include <iostream>
#include <deque>
#include <cctype>
#include <algorithm>


#define rep(i, a, b) for (auto i = a; i < b; i++)
#define REP(i, a, b) for (auto i = a; i <= b; i++)
#define FAST ios_base::sync_with_stdio(false); cin.tie(NULL);cout.tie(NULL)

using namespace std;

int case_cnt;
int arr_len;
string funcs;
string number_string;
deque<int> arr;

void print_arr(bool is_reverse)
{
    cout << "[";

    if (is_reverse)
    {
        for (int i = arr.size() - 1; i >= 0; i--)
        {
            cout << arr[i];
            if (i != 0)
                cout << ",";
        }
    }
    else
    {
        rep(i, 0, arr.size())
        {
            cout << arr[i];
            if (i != arr.size() - 1)
                cout << ",";
        }
    }
    cout << "]\n";
}

int main(void){
    FAST;
    cin >> case_cnt;

    rep(i, 0, case_cnt){
        arr.resize(0);
        cin >> funcs;
        cin >> arr_len;
        cin >> number_string;

        int temp = 0;
        bool is_reverse = false;
        bool is_abrupt = false;

        rep(j, 0, number_string.length()){
            // cout << number_string[j];
            if (isdigit(number_string[j])){
                temp *= 10;
                temp += (number_string[j] - '0');
            }
            else{
                if(temp)
                    arr.push_back(temp);
                temp = 0;
            }
        }

        rep(j, 0, funcs.length()){
            if (funcs[j] == 'R'){
                is_reverse = !is_reverse;
            }
            else{
                if (arr.empty()){
                    is_abrupt = true;
                    break;
                }

                if (is_reverse)
                    arr.pop_back();
                else
                    arr.pop_front();
            }
        }
        if(is_abrupt) cout << "error\n";
        else print_arr(is_reverse);
    }
    
}
